### PR TITLE
Implemented combine retention policy.

### PR DESCRIPTION
**Pull request by ibuenros**
Used to combine different simple retention policies with UNION or INTERSECT operations.

### CODE REVIEW

**Comment by ibuenros** [comment]
@ydailinkedin can you review? Thanks!

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/CombineRetentionPolicy.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+ * </p>
+ */
+public class CombineRetentionPolicy implements RetentionPolicy<DatasetVersion> {
+
+  public static final String RETENTION_POLICIES_PREFIX = DatasetCleaner.CONFIGURATION_KEY_PREFIX +
+      "combine.retention.policy.class.";
+  public static final String DELETE_SETS_COMBINE_OPERATION = DatasetCleaner.CONFIGURATION_KEY_PREFIX +
+      "combine.retention.policy.delete.sets.combine.operation";
+
+  public enum DeletableCombineOperation {
```
Comment (ydailinkedin) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/CombineRetentionPolicy.java]: Are we going to have some priority based combination rule?

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/CombineRetentionPolicy.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
+package gobblin.data.management.retention.policy;
```
Comment (ydailinkedin) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/CombineRetentionPolicy.java]: Missing license header for all files in this commit.

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/UnixTimestampVersionFinder.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return TimestampedDatasetVersion.class;
+  }
+
+  @Override public Path globVersionPattern() {
+    return this.embeddedFinder.globVersionPattern();
+  }
+
+  @Override public TimestampedDatasetVersion getDatasetVersion(Path pathRelativeToDatasetRoot, Path fullPath) {
+    StringDatasetVersion version = this.embeddedFinder.getDatasetVersion(pathRelativeToDatasetRoot, fullPath);
+    Long timestamp = Long.parseLong(version.getVersion());
```
Comment (ydailinkedin) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/UnixTimestampVersionFinder.java]: Is it the case that version.getVersion() always return a long in milliseconds?

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/CombineRetentionPolicy.java (hunk 1 above)
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/CombineRetentionPolicy.java]: Sure, I can add that. What exactly does priority based mean?

File: gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/UnixTimestampVersionFinder.java (hunk 3 above)
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/retention/version/finder/UnixTimestampVersionFinder.java]: You're right, I'll fix this.

**Pull request merged**